component/common/loki/wal/watcher_test.go

```diff
@@ -52,6 +52,26 @@ func (t *testWriteTo) AppendEntries(entries wal.RefEntries, _ int) error {
 	return nil
 }
 
+func (t *testWriteTo) AssertContainsLines(tst *testing.T, lines ...string) {
+	seen := map[string]bool{}
+	for _, l := range lines {
+		seen[l] = false
+	}
+	for _, e := range t.ReadEntries.StartIterate() {
+		if _, ok := seen[e.Line]; ok {
+			seen[e.Line] = true
+		}
+	}
+	t.ReadEntries.DoneIterate()
+
+	allSeen := true
+	for _, wasSeen := range seen {
+		allSeen = allSeen && wasSeen
```

Comment:
nit: simplify to moving the require.True here for wasSeen, could add the line that was missing for better context.